pkg/apis/core/validation/validation.go

```diff
@@ -2169,7 +2172,9 @@ func ValidatePersistentVolumeClaimUpdate(newPvc, oldPvc *core.PersistentVolumeCl
 	}
 
 	if validateStorageClassUpgrade(oldPvcClone.Annotations, newPvcClone.Annotations,
```

Comment:
Better rename this to validateStorageClassUpgradeFromAnnotation

Reply:
Done.